File: 704_binary_search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        left = 0
        right = len(nums) - 1

        while left <= right:
            middle = (left + right) // 2

            if target < nums[middle]:
                right = middle - 1
            elif target == nums[middle]:
                return middle
            else:
                left = middle + 1
        return -1

File: 704-binary-search.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var search = function (nums, target) {
  left = 0;
  right = nums.length - 1;

  while (left <= right) {
    mid = Math.floor((left + right) / 2);
    if (nums[mid] < target) {
      left = mid + 1;
    } else if (nums[mid] == target) {
      return mid;
    } else {
      right = mid - 1;
    }
  }
  return -1;
};
